js/bytetrade.js

```diff
@@ -0,0 +1,1000 @@
+                'fees': '0.0008',
+            },
+            'api': {
+                'market': {
+                    'get': [
+                        'klines',        // Kline of a symbol
+                        'depth',         // Market Depth of a symbol
+                        'trades',        // Trade records of a symbol
+                        'tickers',
+                    ],
+                },
+                'public': {
+                    'get': [
+                        'symbols',        // Reference information of trading instrument, including base currency, quote precision, etc.
+                        'currencies',     // The list of currencies available
+                        'balance',        // Get the balance of an account
+                        'orders/open',    // Get the open orders of an account
+                        'orders/closed',  // Get the closed orders of an account
+                        'orders',         // Get the details of an order of an account
+                        'orders/trades',  // Get detail match results
+                        'depositaddress', // Get deposit address
+                        'withdrawals',    // Get withdrawals info
+                        'deposits',       // Get deposit info
+                        'transfers',      // Get transfer info
+                        'transaction',    // Post create order\cancel order\withdraw\transfer transaction to blockchain
+                    ],
+                    'post': [
+                    ],
+                },
+                'private': {
+                    'get': [
+                    ],
+                    'post': [
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'taker': 0.0008,
+                    'maker': 0.0008,
+                },
+            },
+            'commonCurrencies': {
+                'CMT@18': 'CMT@18',
+            },
+            // exchange-specific options
+            'options': {
+            },
+            'exceptions': {
+                'vertify error': AuthenticationError,           // private key signature is incorrect
+                'transaction already in network': BadRequest,   // same transaction submited
+                'invalid argument': ArgumentsRequired,
+            },
+        });
+    }
+
+    async fetchCurrencies (params = {}) {
+        const currencies = await this.publicGetCurrencies (params);
+        const result = {};
+        for (let i = 0; i < currencies.length; i++) {
+            const currency = currencies[i];
+            const id = this.safeString (currency, 'code');
+            const name = this.safeValue (currency, 'name');
+            const nameId = name + '@' + id;
+            if (!(nameId in this.commonCurrencies)) {
+                this.commonCurrencies[nameId] = name;
```

Comment:
this is not allowed, we don't want to mutate `commonCurrencies`,

also please use `safeCurrencyCode` instead of `commonCurrencyCode`

Reply:
thanks a lot for your reveiw.
in bytetrade.com DEX, request [https:api-v2.bytetrade.com/currencies](url) will return currencies info,
the api doc is [https:github.com/Bytetrade/bytetrade-official-api-docs/wiki/rest-api#get-currencies-get-currencys-supported-in-bytetradecom](url)

we can see the coin name is none-unique in the result, the coin which code is 18 is the  CyberMiles ERC20, and the coin which code is 35 is the CyberMiles main chain, but their name is same.
that is because bytetrade is a DEX, supports people create coin with the same name, but the id(code) of coin is unique, so we should use the id or name and id as the identity of coin.
For coin name and symbol is same with CCXT, I use name@id as the key of commonCurrencies dict.
If the name@id is in commonCurrencies, safeCurrencyCode() call will return the commonCurrencies[name@id], for example, CMT@18 will return code CMT@18;
If the name@id is not in commonCurrencies, 'commonCurrencies[name@id] = name' will be set in fetchCurrencies function, after that, safeCurrencyCode() call will return correct code, for example CMT@35 will return CMT.

I have tried many other methods, but they can not achieve the above effect. If you can have a better way, please let me know, thank you.

             here is part of currencies result:
             [{
                 "name": "CMT",       currency name, non-unique
                 "code": "18",        currency id, unique
                 "type": "crypto",
                 "fullname": "CyberMiles",
                 "active": true,
                 "chainType": "ethereum",
                 "basePrecision": 18,
                 "transferPrecision": 10,
                 "externalPrecision": 18,
                 "chainContractAddress": "0xf85feea2fdd81d51177f6b8f35f0e6734ce45f5f",
                 "limits": {
                   "deposit": {
                     "min": "0",
                     "max": "-1"
                   },
                   "withdraw": {
                     "min": "0",
                     "max": "-1"
                   }
                 }
               },
               {
                 "name": "CMT",
                 "code": "35",
                 "type": "crypto",
                 "fullname": "CyberMiles",
                 "active": true,
                 "chainType": "cmt",
                 "basePrecision": 18,
                 "transferPrecision": 10,
                 "externalPrecision": 18,
                 "chainContractAddress": "0x0000000000000000000000000000000000000000",
                 "limits": {
                   "deposit": {
                     "min": "1",
                     "max": "-1"
                   },
                   "withdraw": {
                     "min": "10",
                     "max": "-1"
                   }
                 }
               }
               ]

Reply:
@frosty00, i get the right way to use commonCurrencies, please help review again, thank you.